src/Microsoft.Azure.WebJobs.Host/Executors/StorageAccountProviderExtensions.cs

```diff
@@ -54,7 +56,25 @@ public static Task<IStorageAccount> GetStorageAccountAsync(this IStorageAccountP
                 }
             }
 
-            return provider.GetAccountAsync(connectionStringName, cancellationToken);
+            IStorageAccount account = await provider.GetAccountAsync(connectionStringName, cancellationToken);
+            ValidateStorageAccount(account, connectionStringName);
+            return account;
+        }
+
+        private static void ValidateStorageAccount(IStorageAccount account, string connectionStringName)
```

Comment:
GetAccountAsync also handles dashboard storage account. To avoid that will have to introduce additional logic which is already separated by https://github.com/Azure/azure-webjobs-sdk/pull/946/files#diff-4bab545de76b34702affefd1e5180ae9L15